--- scli-0.2.11/configure.in~	Fri Aug 30 14:37:07 2002
+++ scli-0.2.11/configure.in	Fri Aug 30 14:37:07 2002
@@ -59,10 +59,7 @@
 
 dnl Checks for library functions.
 
-AC_CHECK_FUNC(gethostbyname)
-if test $ac_cv_func_gethostbyname = no; then
-  AC_CHECK_LIB(nsl, gethostbyname)
-fi
+AC_CHECK_FUNC(gethostbyname, [], AC_CHECK_LIB(nsl, gethostbyname))
 
 AC_CHECK_FUNC(connect)
 if test $ac_cv_func_connect = no; then
@@ -76,8 +73,8 @@
 
 AC_ARG_ENABLE(dmalloc,
 [  --enable-dmalloc        enable dmalloc debugging (www.dmalloc.com)],
-  AC_CHECK_HEADERS(dmalloc.h)
-  AC_CHECK_LIB(dmalloc, malloc) 
+  [AC_CHECK_HEADERS(dmalloc.h)
+  AC_CHECK_LIB(dmalloc, malloc)]
 )
 
 dnl Create all the Makefiles we need and we are done.
